src/handlers/token_handler.erl

```diff
@@ -0,0 +1,32 @@
+-module(token_handler).
+
+-export([init/3,
+         allowed_methods/2,
+         content_types_provided/2,
+         is_authorized/2,
+         to_json/2]).
+
+init(_Transport, _Req, []) ->
+    {upgrade, protocol, cowboy_rest}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>], Req, State}.
+
+is_authorized(Req, State) ->
+    case cowboy_req:parse_header(<<"authorization">>, Req) of
+        {ok, {<<"basic">>, {Email, Password}}, Req2} ->
+            case db_user:authenticate(Email, Password) of
+                {ok, User} ->
+                    {true, Req2, #{user => User}};
+                _ -> {{false, <<"Basic realm=\"holidayping\"">>}, Req2, State}
```

Comment:
So that was to say this actual bit probably won't be repeated elsewhere :P the token auth of the rest of the handlers probably will be repeated, so there I'll use an util function or that mixer which looks pretty useful.